src/components/Pagination/Pagination.tsx

```diff
@@ -1 +1,98 @@
-export const Pagination = () => {};
+import classNames from 'classnames';
+import React from 'react';
+
+type PaginationProps = {
+  total: number;
+  perPage: number;
+  currentPage: number;
+  onPageChange: (page: number) => void;
+};
+
+export const Pagination: React.FC<PaginationProps> = ({
+  total,
+  perPage,
+  currentPage,
+  onPageChange,
+}) => {
+  const arrayOfPages = Array.from(
+    { length: Math.ceil(total / perPage) },
+    (_, i) => i + 1,
+  );
+
+  const arrayOfItems = Array.from(
+    { length: Math.min(perPage, total - (currentPage - 1) * perPage) },
+    (_, i) => {
+      return `Item ${i + 1 + (currentPage - 1) * perPage}`;
+    },
+  );
+
+  return (
+    <div>
+      <ul className="pagination">
+        <li
+          className={classNames(`page-item`, { disabled: currentPage === 1 })}
+        >
+          <a
+            data-cy="prevLink"
+            className="page-link"
+            href="#prev"
+            aria-disabled={currentPage === 1}
+            onClick={() => {
+              if (currentPage > 1) {
+                onPageChange(currentPage - 1);
+              }
+            }}
+          >
+            «
+          </a>
+        </li>
+        {arrayOfPages.map(page => (
+          <li
+            key={page}
+            className={classNames('page-item', {
+              active: page === currentPage,
+            })}
+          >
+            <a
+              data-cy="pageLink"
+              className="page-link"
+              href={`#${page}`}
+              onClick={() => {
+                onPageChange(page);
+              }}
```

Comment:
To prevent the default navigation behavior of the anchor tag, accept the event parameter in the `onClick` handler and call `event.preventDefault()`. This will stop the URL from changing.